src/controller/CHIPDevice.cpp

```diff
@@ -69,7 +69,7 @@ CHIP_ERROR Device::SendMessage(System::PacketBuffer * buffer)
         resend = buffer;
     }
 
-    err    = mSessionManager->SendMessage(mDeviceId, buffer);
+    err    = mSessionManager->SendMessage(mSecureSession, buffer);
```

Comment:
@kghost , how does the value of `mSecureSession` get populated when the device is already paired, and it is `deserialized` from the persistent storage?

Reply:
I'm not sure `SecureSessionHandle` should be able to be serialized to persistent storage. There will be several ways to obtain a handle.

Messaging layer:
1. Server: From `SecureSessoinDelegate::OnNewConnection`
2. Client: Not sure, while looking the newest code, `SecurePairingSession::DeriveSecureSession` takes my attention

Application interface (Messaging layer API):
1. Server: Register unsolicited message handler, then from `ExchangeDelegate::OnMessageReceived` of that handler
2. Client: Using `Channel` API, https://github.com/project-chip/connectedhomeip/pull/4019

Reply:
I suspect, the current PR will break the communication between the controller and the device. After pairing is complete, the controller saves the device information in the persistent storage. When a message is sent to the device, the controller creates the `CHIPDevice` object by reading it from the storage. With the current PR, the created (deserialized) device object will not have a valid `mSecureSession` object.

Will you be making more changes in the current PR to fix this issue?

Reply:
There will be a Channel API to create the `mSecureSession` object. But Channel implementation depends on this PR. So to break to loop, we can temporary create an `mSecureSession` from `PeerNodeId`.